Source/WebKit/UIProcess/mac/WebPageProxyMac.mm

```diff
@@ -237,7 +238,8 @@ static inline bool expectsLegacyImplicitRubberBandControl()
     if (!hasRunningProcess())
         return false;
 
-    grantAccessToCurrentPasteboardData(pasteboardName);
+    if (auto replyID = grantAccessToCurrentPasteboardData(pasteboardName, [] () { }))
+        websiteDataStore().protectedNetworkProcess()->connection()->waitForAsyncReplyAndDispatchImmediately<Messages::NetworkProcess::AllowFilesAccessFromWebProcess>(*replyID, 100_ms);
```

Comment:
There's already a hang in this function below and the below sync call won't work if this one has not completed.

Reply:
Yeah, I don't think this really makes the problem _worse_ (I do agree this sync IPC isn't ideal, though I'm not sure how much we can do about it, since it's public API we need to continue supporting).

I also imagine that sync IPC between the network process and UI process (both of which should not be compromised by maliciously crafted content) is much less bad than blocking on the untrusted web content process. A short 100 ms timeout seems reasonable.

Reply:
The thing I'd actually be more concerned about is cross-process IPC deadlocks — for instance, the UI process waiting for the network process while the network process simultaneously syncwaits for the UI process. Hopefully this isn't an issue, and the sync IPC only goes in one direction...

Reply:
To close the loop above. I think the only non-test message that the network process synchronously sends the UI process / waits for is `SecItemShimProxy::SecItemRequestSync` which I think would never be called during a `AllowFilesAccessFromWebProcess.`